### PR TITLE
DSC19-39 create hmis package

### PR DESCRIPTION
**Pull request by bradsawadye**
This creates an instance of DHIS

### CODE REVIEW

File: health-management-information-system/docker/configure-docker.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (MattyJ007) [inline health-management-information-system/docker/configure-docker.sh:1]: Why is this file here?

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
You have some weird files hanging about

File: health-management-information-system/instant.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "hmis",
+  "name": "hmsi",
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/instant.json]: addressed

File: health-management-information-system/instant.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  "description": "",
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/instant.json]: addressed

File: health-management-information-system/instant.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  "type": "infrastructure",
+  "version": "v1.0.0",
+  "dependencies": []
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/instant.json]: addressed

File: health-management-information-system/package.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/package.json]: addressed

File: health-management-information-system/.prettierrc.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+semi: false
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/.prettierrc.yml]: addressed

File: health-management-information-system/.eslintrc.yml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+env:
```
[the remark this answers is not in the document]
Comment (bradsawadye) [reply health-management-information-system/.eslintrc.yml]: addressed

File: health-management-information-system/docker/configure-docker.sh (hunk 1 above)
Comment (bradsawadye) [reply health-management-information-system/docker/configure-docker.sh]: addressed

**Review by MattyJ007 — COMMENTED** [review]
@bradsawadye Looking good - one small thing could you delete the empty README files if there's nothing valuable to go in them.

File: health-management-information-system/kubernetes/main/k8s.sh [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+k8sMainRootFilePath=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+
```
Comment (MattyJ007) [inline health-management-information-system/kubernetes/main/k8s.sh:4]: @bradsawadye your k8s service is working well 👍🏼 One improvement though would be to print out the service URLs for minikube in the script - it really makes testing easier. The core script outputs the URLs really well: https://github.com/openhie/instant/blob/master/core/kubernetes/main/k8s.sh

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
Looking good. One small suggestion

File: health-management-information-system/kubernetes/main/k8s.sh (hunk 8 above)
Comment (bradsawadye) [reply health-management-information-system/kubernetes/main/k8s.sh:4]: addresed

File: health-management-information-system/README.md [hunk 9]
```diff
@@ -1,3 +1,80 @@
...
+
+To access DHIS
+
+### In Docker
+
+* URL - `http://localhost:8081`
+
+### In Kubernetes
+
+Use the host ip or DNS, and the port `8081`
```
Comment (MattyJ007) [inline health-management-information-system/README.md:75]: @bradsawadye This is not the case on Minikube

Comment (bradsawadye) [reply health-management-information-system/README.md]: addressed

File: .env.local [hunk 10]
```diff
@@ -4,3 +4,10 @@ OPENHIM_TRANSACTION_API_PORT=5001
 OPENHIM_MEDIATOR_API_PORT=8080
 MOCK_SERVER_HOST=localhost
 MOCK_SERVER_PORT=4000
+
+# DHIS details
+DHIS2_PROTOCOL=http
+DHIS2_API_HOSTNAME=localhost
```
Comment (MattyJ007) [inline .env.local:10]:
```suggestion
DHIS2_API_HOSTNAME=<dhis2_hostname>
```

File: .env.local [hunk 11]
```diff
@@ -4,3 +4,10 @@ OPENHIM_TRANSACTION_API_PORT=5001
...
+DHIS2_API_PASSWORD=district
+DHIS2_API_PORT=8081
```
Comment (MattyJ007) [inline .env.local:12]:
```suggestion
DHIS2_API_PORT=<dhis2_port>
```

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
Looking good - tests all pass 👍🏼
Please could you add something to the main readme about how to run the tests

File: .env.remote [hunk 12]
```diff
@@ -4,3 +4,10 @@ OPENHIM_TRANSACTION_API_PORT=5001
 OPENHIM_MEDIATOR_API_PORT=<openhim_api_port>
 MOCK_SERVER_HOST=<mock_service_hostname>
 MOCK_SERVER_PORT=4000
+
+# DHIS details
+DHIS2_PROTOCOL=http
+DHIS2_API_HOSTNAME=localhost
```
Comment (MattyJ007) [inline .env.remote:10]: @bradsawadye Did I put my comment on the wrong file?

**Review by MattyJ007 — APPROVED** [review]
LGTM 👍🏼
